pkg/cstorvolumereplica/v1alpha1/cstorvolumereplica_test.go

```diff
@@ -0,0 +1,206 @@
+			[]string{
+				"uid1",
+			},
+		},
+		"Present 2": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}}},
+			},
+			[]string{
+				"uid1",
+				"uid2",
+			},
+		},
+		"Present 3": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid3"}}}},
+			},
+			[]string{
+				"uid1",
+				"uid2",
+				"uid3",
+			},
+		},
+		"Present 4": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid3"}}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{cstorPoolUIDLabelKey: "uid4"}}}},
+			},
+			[]string{
+				"uid1",
+				"uid2",
+				"uid3",
+				"uid4",
+			},
+		},
+		// UIDS are not present
+		"Not Present 1": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+			},
+			[]string{
+				"",
+			},
+		},
+		"Not Present 2": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+			},
+			[]string{
+				"",
+				"",
+			},
+		},
+		"Not Present 3": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+			},
+			[]string{
+				"",
+				"",
+				"",
+			},
+		},
+		"Not Present 4": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+			},
+			[]string{
+				"",
+				"",
+				"",
+				"",
+			},
+		},
+	}
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			cvr := &cvrList{items: mock.cvrs}
+			if output := cvr.GetPoolUIDs(); !reflect.DeepEqual(output, mock.expectedString) {
+				t.Fatalf("test %q failed: expected %v \n got : %v \n", name, mock.expectedString, output)
+			}
+		})
+	}
+}
+
+func TestWithListObject(t *testing.T) {
+	tests := map[string]struct {
+		expectedCvrs []apis.CStorVolumeReplica
+	}{
+		"Object 1": {
+			[]apis.CStorVolumeReplica{
```

Comment:
Please change the table to include only the primitive data types

Reply:
done